### DISCUSSION TITLE
Change decorator at runtime

### DISCUSSION THREAD

**Discussion by thefrana** [fastify/fastify#3145] [Q&A] [answered]
Hi,
I am using [MikroORM](https://mikro-orm.io/) for my database. It uses IdentityMap and it requires [forking entity manager on every request](https://mikro-orm.io/docs/identity-map/#why-is-request-context-needed). I tried it, but it throws an error that I cannot change decorator at runtime (or override it).

```ts
app.addHook('onRequest', async () => {
   app.decorate('em', app.orm.em.fork())
})
```

Can you help me out? 

Thanks

**Comment by zekth** [comment, marked as answer]
just do
```js
app.addHook('onRequest', async (request, reply) => {
   request.em = app.orm.em.fork()
})
```

**Reply by thefrana** [reply]
Thank you for quick response!

**Comment by jsumners** [comment]
```js
app.addHook('onRequest', async () => {
   app.decorate('em', app.orm.em.fork())
})
```

This is saying "on each request add a new decoration". That is not how decorators are meant to be used. A decoration should be added to the request to define the shape and then that decoration can be used as @zekth outlined.

Please read https://www.fastify.io/docs/latest/Decorators/

**Reply by thefrana** [reply]
Thank you, it helped
